core/src/main/java/org/eclipse/hono/util/AddressHelper.java

```diff
@@ -0,0 +1,98 @@
+/*******************************************************************************
+ * Copyright (c) 2020 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+package org.eclipse.hono.util;
+
+import java.util.Objects;
+import java.util.regex.PatternSyntaxException;
+
+/**
+ * Utility class for handling AMQP addresses.
+ */
+public class AddressHelper {
+
+    static final String ID_REGEX = "[a-zA-Z0-9-_\\.]+";
```

Comment:
I think it makes sense in this case to pre-compile the regular expression using `Pattern`.

Reply:
@ctron It's not a pattern at this point, but a substitute that can be used when writing rules, so you can write `telemetry/${ID}` instead of `telemetry/[a-zA-Z0-9-_\\.]+`. It's also moved now to the other place per Kai's suggestion.